src/main/java/de/caritas/cob/agencyservice/api/service/TenantHibernateInterceptor.java

```diff
@@ -13,16 +15,52 @@
 @Slf4j
 public class TenantHibernateInterceptor extends EmptyInterceptor {
 
+  private static Boolean multiTenancyEnabled;
+
   @Override
   public void preFlush(Iterator entities) {
-    Object entity;
-    while (entities.hasNext()) {
-      entity = entities.next();
-      if (entity instanceof TenantAware) {
-        ((TenantAware) entity).setTenantId(TenantContext.getCurrentTenant());
+
+    if (multiTenancyEnabled == null) {
+      multiTenancyEnabled = resolveMultiTenancy();
+    }
+
+    if (multiTenancyEnabled) {
+      Object entity;
+      while (entities.hasNext()) {
+        entity = entities.next();
+        if (entity instanceof TenantAware) {
+          ((TenantAware) entity).setTenantId(TenantContext.getCurrentTenant());
+        }
       }
     }
 
     super.preFlush(entities);
   }
+
+  private boolean resolveMultiTenancy() {
```

Comment:
please extract to separate class like MultitenancyPropertyResolver

Reply:
Implementation of this class is really specific to this hibernate interceptor, since it can not be used with spring context. I don't see any use case where the reading of spring properties should be done like this. So I prefer to have it to not separate since we would then need to copy these 2 classes to other microservices instead of just copying it together in one class and this impl.  could  also be misused on other places if it would be in a seperate class.